core-backend/common/src/lib.rs

```diff
@@ -129,22 +123,19 @@ pub trait Environment<E: Ext + IntoExtInfo + 'static>: Sized {
 }
 
 pub trait OnSuccessCode<T, E> {
-    fn on_success_code<F>(self, f: F) -> Result<i32, String>
+    fn on_success_code<F>(self, f: F) -> Result<i32, E>
     where
         F: FnMut(T) -> Result<(), E>;
 }
 
-impl<T, E, E2> OnSuccessCode<T, E2> for Result<T, E>
-where
-    E2: Into<String>,
-{
-    fn on_success_code<F>(self, mut f: F) -> Result<i32, String>
+impl<T, E> OnSuccessCode<T, E> for Result<T, E> {
+    fn on_success_code<F>(self, mut f: F) -> Result<i32, E>
     where
-        F: FnMut(T) -> Result<(), E2>,
+        F: FnMut(T) -> Result<(), E>,
     {
         match self {
             Ok(t) => {
```

Comment:
```suggestion
            Ok(t) => f(t).map(|_| 0),
```

Reply:
done in [0817d34](https://github.com/gear-tech/gear/pull/853/commits/0817d34d8fe8168d3d646a8c27dd205e507d585b)